nbodykit/algorithms/pair_counters/mocksurvey.py

```diff
@@ -0,0 +1,175 @@
+from .base import PairCountBase, verify_input_sources
+import numpy
+import logging
+
+
+class SurveyDataPairCount(PairCountBase):
+    r"""
+    Count (weighted) pairs of objects from a survey data catalog using the
+    :mod:`Corrfunc` package and a 1D, 2D, projected, or angular coordinate basis.
+
+    See the Notes below for the allowed coordinate dimensions
+
+    Results are computed when the class is inititalized. See the documenation
+    of :func:`~SurveyDataPairCount.run` for the attributes storing the
+    results.
+
+    .. note::
+
+        The algorithm expects the positions of particles from a survey catalog
+        be the sky coordinates, right ascension and declination, and redshift.
+        To compute pair counts in a simulation box using Cartesian
+        coordinates, see :class:`~nbodykit.algorithms.SimulationBoxPairCount`.
+
+    .. warning::
+        The right ascension and declination columns should be specified
+        in degrees.
+
+    Parameters
+    ----------
+    mode : '1d', '2d', 'projected', 'angular'
+        compute pair counts as a function of the specified coordinate basis;
+        see the Notes section below for specifics
+    first : CatalogSource
+        the first source of particles, providing the 'Position' column
+    edges : array_like
+        the separation bin edges along the first coordinate dimension;
+        depending on ``mode``, the options are :math:`r`, :math:`r_p`, or
+        :math:`\theta`. Expected units for distances are :math:`\mathrm{Mpc}/h`
+        and degrees for angles. Length of nbins+1
+    cosmo : :class:`~nbodykit.cosmology.cosmology.Cosmology`, optional
```

Comment:
What if the fiducial cosmology doesn't have radiation?

Reply:
Do we assume that somewhere? It is only used to compute comoving distance to pass to Corrfunc

Reply:
Class can't do without radiation..

Reply:
oh sure, but that's a more general issue with the Cosmology class. We should raise a more helpful message when Tcmb0 is 0 in Cosmology?